File: CppPractice/Project1/Project1/Leetcode119.cpp
#include "LeetcodeHeader.h"

class Solution {
public:
	vector<int> getRow(int rowIndex) {
		vector<vector<int>> triangle;
		vector<int> temp;
		if (rowIndex == 0)
		{
			return { 1 };

		}
		else if (rowIndex == 1)
		{
			return { 1,1 };
		}

		else
		{
			temp = { 1,1 };
			triangle.push_back(temp);
			for (int i = 2; i <= rowIndex; i++)
			{
				temp = {};
				for (int j = 0; j <= i; j++)
				{
					if (j == 0 or j == i)
					{
						temp.push_back(1);
					}
					else
					{
						temp.push_back(triangle.back()[j - 1] + triangle.back()[j]);
					}
				}
				triangle.push_back(temp);
			}

			return temp;
		}

	}
};

File: CSharpPractice/LeetcodePractice/Leetcode119.cs
public class Solution119 {
    public IList<int> GetRow(int rowIndex) 
    {
        List<int> temp = new List<int>{1};
        List<int> ans = new List<int>{1,1};
        if(rowIndex==0){return temp;}
        if(rowIndex==1){return ans;}
        
        for(int i=0;i<rowIndex-1;i++)
        {
            temp = new List<int>{1};
            for(int j =1;j<i+2;j++)
            {
                temp.Add(ans[j-1]+ans[j]);
            }
            temp.Add(1);
            ans=temp;
        }
        return ans;        
    }
}